utils/cautils/token_generator.go

```diff
@@ -387,3 +397,56 @@ func generateJWKToken(ctx *cli.Context, p *provisioner.JWK, tokType int, tokAttr
 		return tokenGen.Token(tokAttrs.subject)
 	}
 }
+
+func generateRenewToken(ctx *cli.Context, aud, sub string) (string, error) {
+	renewCert := ctx.String("x5c-cert")
+	if renewCert == "" {
+		return "", errs.RequiredFlag(ctx, "x5c-cert")
+	}
+	renewKey := ctx.String("x5c-key")
+	if renewKey == "" {
+		return "", errs.RequiredFlag(ctx, "x5c-key")
+	}
+
+	bundle, err := pemutil.ReadCertificateBundle(renewCert)
+	if err != nil {
+		return "", err
+	}
+	if len(bundle) == 0 {
+		return "", errs.InvalidFlagValueMsg(ctx, "--x5c-cert", renewCert, "certificate not found")
+	}
+	key, err := pemutil.Read(renewKey)
+	if err != nil {
+		return "", err
+	}
+	if sub != "" && sub != bundle[0].Subject.CommonName {
+		return "", errors.Errorf("positional argument <subject> must match the certificate common name")
+	}
+
+	var issuer string
+	if ext, ok := provisioner.GetProvisionerExtension(bundle[0]); ok {
```

Comment:
What happens if/when the provisioner extension does not exist? (Vault RA)

Reply:
Fixed with 4a1848d